src/openfermion/ops/_symbolic_operator.py

```diff
@@ -568,16 +588,32 @@ def __eq__(self, other):
             a = self.terms[term]
             b = other.terms[term]
             # math.isclose does this in Python >=3.5
-            if not abs(a - b) <= max(EQ_TOLERANCE,
-                                     EQ_TOLERANCE * max(abs(a), abs(b))):
-                return False
+            if isinstance(a, sympy.Expr) or isinstance(b, sympy.Expr):
+                # Can only use absolute inequality here as max(1, a, b)
+                # is not necessarily well-defined for expressions.
+                if sympy.simplify(abs(a - b) <= EQ_TOLERANCE) != True:
+                    return False
+            else:
+                if not abs(a - b) <= EQ_TOLERANCE * max(1, abs(a), abs(b)):
+                    return False
         # terms only in one (compare to 0.0 so only abs_tol)
         for term in set(self.terms).symmetric_difference(set(other.terms)):
             if term in self.terms:
-                if not abs(self.terms[term]) <= EQ_TOLERANCE:
-                    return False
-            elif not abs(other.terms[term]) <= EQ_TOLERANCE:
-                return False
+                if isinstance(self.terms[term], sympy.Expr):
+                    if sympy.simplify(
+                            abs(self.terms[term]) <= EQ_TOLERANCE) != True:
```

Comment:
The code for checking whether a sympy Expr is small is repeated many times. Factor it out into a helper function.

Reply:
Fair point, I'll stop being lazy